From bf12c77b6e1cc5b7b0d42e26a58d1fc45378176e Mon Sep 17 00:00:00 2001
From: Piotr Zierhoffer <pzierhoffer@antmicro.com>
Date: Thu, 17 Feb 2022 16:40:38 +0100
Subject: [PATCH 1/2] Change socket port to not conflict with Robot defaults

---
 emulation/xous-release.resc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/emulation/xous-release.resc b/emulation/xous-release.resc
index 05f318a7..d5965d1b 100644
--- a/emulation/xous-release.resc
+++ b/emulation/xous-release.resc
@@ -87,7 +87,7 @@ macro reset
 
 # Connect the serial port to an Ethernet port
 mach set 0
-emulation CreateServerSocketTerminal 9999 "kernel" False
+emulation CreateServerSocketTerminal 8888 "kernel" False
 connector Connect sysbus.uart kernel
 
 runMacro $reset
-- 
2.33.0

